### DISCUSSION TITLE
ScriptReference/EditorGUI.ProgressBar

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#623] [Notes]
# ScriptReference/EditorGUI.ProgressBar

https://docs.unity3d.com/ScriptReference/EditorGUI.ProgressBar.html

**Comment by naelstrof** [comment]
This example doesn't work great if you're using EditorGUILayout, a wizard, or an inspector as it ignores other elements.

The following will make it play nicely with EditorGUILayout:
```csharp
var rect = EditorGUILayout.BeginVertical();
EditorGUI.ProgressBar(rect, 0.5, "Halfway");
GUILayout.Space(20);
EditorGUILayout.EndVertical();
```

**Reply by rob5300** [reply]
`EditorGUILayout.GetControlRect()` can be used with ProgressBar() so you can use it in Horizontal/Vertical layouts.
